tests/test_assemble_coadd.py

```diff
@@ -334,6 +336,52 @@ def testInputMap(self):
             else:
                 self.assertTrue(np.all(inputMap.check_bits_pix(validPix, [visitBitDict[expId]])))
 
+    @lsst.utils.tests.methodParameters(doOnlineForMean=[False, True])
+    def testArtifactMask(self, doOnlineForMean):
+        """Test that CompareWarp and AssembleCoadd with artifact mask produce
+        identical results.
+        """
+        config = MockCompareWarpAssembleCoaddConfig()
+        config.doOnlineForMean = doOnlineForMean
+        compareWarpTask = MockCompareWarpAssembleCoaddTask(config=config)
+        compareWarpResult = compareWarpTask.runQuantum(
+            self.skyInfo,
+            self.dataRefList,
+            psfMatchedWarpRefList=self.dataRefListPsfMatched,
+        )
+
+        # Make a new list of dataRefs with artifact masks applied.
+        dataRefList = []
+        for dataRef, artifactMask in zip(self.dataRefList, compareWarpResult.altMaskList):
+            dataSet = dataRef.get()
+            compareWarpTask.applyAltMaskPlanes(dataSet.mask, artifactMask)
+            # Repackage the dataset into a DataRef.
+            dataRefList.append(
+                pipeBase.InMemoryDatasetHandle(
+                    dataSet, storageClass=dataRef.storageClass, copy=True, dataId=dataRef.dataId
+                )
+            )
```

Comment:
I'd recommend replacing `dataRef` here with `handle` or something, because `DataRef` is a type that no longer exists (it's from the Gen2 butler).  Might even be worth doing that in the task itself (as a separate commit).

Reply:
You mean change the variable name to say `handle` instead of `ref`, right?

Reply:
Yes, exactly.

Reply:
The term `ref` is baked in quite strongly in the main task, sometimes as keyword-only arguments. It would also result in some temporary mismatch in the convention if we don't change the name in some tasks in `pipe_tasks`. So for the moment, I've fixed this in tests and in docstrings. I'm afraid a complete change would have to happen later.